climada/util/coordinates.py

```diff
@@ -1057,6 +1057,79 @@ def assign_coordinates(coords, coords_to_assign, distance="euclidean",
                 coords_to_assign, coords[not_assigned_idx_mask], threshold, **kwargs)
     return assigned_idx
 
+def assign_haz_centroids(coord_df, hazard, crs, distance='euclidean',
+                        threshold=NEAREST_NEIGHBOR_THRESHOLD,
+                        overwrite=True):
+    """Assign to each exposure's coordinate point its closest hazard's coordinate.
+    If disatances > threshold in points' distances, -1 is returned.
+    If raster hazard and coordinate point outside raster, -1 is returned.
+
+    Parameters
+    ----------
+    coord_df : pd.DataFrame
+        Dataframe with defined latitude/longitude column
+    hazard : Hazard
+        Hazard to match (with raster or vector centroids).
+    crs : pyproj.crs.crs.CRS
+        Coordinate system of the dataframe
+    distance : str, optional
+        Distance to use in case of vector centroids.
+        Possible values are "euclidean", "haversine" and "approx".
+        Default: "euclidean"
+    threshold : float
+        If the distance (in km) to the nearest neighbor exceeds `threshold`,
+        the index `-1` is assigned.
+        Set `threshold` to 0, to disable nearest neighbor matching.
+        Default: 100 (km)
+    overwrite: bool
+        If True, overwrite centroids already present. If False, do
+        not assign new centroids. Default is True.
```

Comment:
I'd rather return a data frame or even just a column of a data frame instead of messing around with (= modifying) input data. It will just cause headaches.

Reply:
I did the method so the behaviour of exp.assign_centroid() remains unchanged, since this operation is used all the time in climada. I think if we change this behaviour, we would have to have bit bigger discussion?

Could it be an option to change the default to overwrite=False in this function, but always pass overwrite=True from the exp.assign_centroids() method?

Reply:
I agree, `exp.assign_centroid()` should remain unchanged in behavior atm. But nevertheless should this method be free from side effects. The solution is to deal with the `overwrite` argument in `assign_centroid()`. 
Little more complexity there - a little less here - better distributed. 😁